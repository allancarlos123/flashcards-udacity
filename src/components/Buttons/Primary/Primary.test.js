import React from 'react';
import { create } from 'react-test-renderer';

import PrimaryButton from './';

describe('PrimaryButton', () => {
  const onPress = jest.fn();
  const tree = create(<PrimaryButton title={'Something'} onPress={onPress} />);
  
  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('calls onPress when pressed', () => {
    tree.root.props.onPress();
    expect(onPress).toHaveBeenCalled();
  });
});