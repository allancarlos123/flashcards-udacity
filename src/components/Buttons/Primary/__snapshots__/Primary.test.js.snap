// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 5,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "borderColor": "#007AFF",
          "borderRadius": 6,
          "borderWidth": 1,
          "overflow": "hidden",
        },
        Object {
          "borderColor": "#ed4e5f",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={undefined}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#ed4e5f",
          },
          Object {
            "paddingVertical": 12,
          },
        ]
      }
    >
      Something
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ed4e5f",
          "height": "100%",
          "opacity": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#FFFFFF",
              "paddingVertical": 12,
            },
            undefined,
            undefined,
          ]
        }
      >
        Something
      </Text>
    </View>
  </View>
</View>
`;
